third_party/terraform/tests/resource_compute_instance_test.go.erb

```diff
@@ -1950,6 +1954,140 @@ func testAccCheckComputeInstanceDestroyProducer(t *testing.T) func(s *terraform.
 	}
 }
 
+func TestComputeInstance_networkIPCustomizedDiff(t *testing.T) {
+	t.Parallel()
+
+	d := &ResourceDiffMock{
+		Before: map[string]interface{}{
+			"network_interface.#": 0,
+		},
+		After: map[string]interface{}{
+			"network_interface.#": 1,
+		},
+	}
+
+	err := forceNewIfNetworkIPNotUpdatableFunc(d);
+
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	if d.IsForceNew {
+		t.Errorf("Expected not force new if network_interface array size static")
+	}
+
+	cases := map[string]struct {
```

Comment:
I'm fine with this as it is, but if you wanted to increase the maintainability somewhat you could create another struct type for NI properties and then have cases contain just Before, After, and ExpectedForceNew

Reply:
refactored